frame/support/src/lib.rs

```diff
@@ -1651,37 +1646,37 @@ pub mod pallet_prelude {
 /// 		Something(u32),
 /// 	}
 ///
-/// 	// Define a struct which implements `frame_support::traits::Get<T::Balance>`
+/// 	// Define a struct which implements `frame_support::traits::Get<T::Balance>` (optional).
 /// 	#[pallet::type_value]
 /// 	pub(super) fn MyDefault<T: Config>() -> T::Balance { 3.into() }
 ///
-/// 	// Declare a storage, any amount of storage can be declared.
+/// 	// Declare a storage item. Any amount of storage items can be declared (optional).
 /// 	//
 /// 	// Is expected either `StorageValue`, `StorageMap` or `StorageDoubleMap`.
-/// 	// The macro generates for struct `$identP` (for storage of name `$ident`) and implement
-/// 	// storage instance on it.
-/// 	// The macro macro expand the metadata for the storage with the type used:
-/// 	// * For storage value the type for value will be copied into metadata
-/// 	// * For storage map the type for value and the type for key will be copied into metadata
-/// 	// * For storage double map the type for value, key1, and key2 will be copied into
+/// 	// The macro generates the prefix type and replace first generic `_`.
+/// 	//
+/// 	// The macro expands the metadata for the storage item with the type used:
+/// 	// * for a storage value the type of the value is copied into the metadata
+/// 	// * for a storage map the type of the values and the type of the key is copied into the metadata
+/// 	// * for a storage double map the types of the values and keys are copied into the
 /// 	//   metadata.
 /// 	//
-/// 	// NOTE: for storage hasher, the type is not copied because storage hasher trait already
+/// 	// NOTE: for storage hasher, the type is not copied because the storage hasher trait already
 /// 	// implements metadata. Thus generic storage hasher is supported.
```

Comment:
```suggestion
/// 	// NOTE: For implementing the storage hasher, the type is not copied because the storage hasher
/// 	//  trait already implements metadata. Thus generic storage hashers are supported.
```
Do you mean something like this?

Reply:
Instead of "For implementing the storage hasher", I would then write "For implementing the storage hasher metadata" or "about storage hasher" or "about storage hasher metadata",

the wording "the trait" isn't very more clear. The full meaning is: the generic `Hasher` is meant to implement `StorageHasher` trait (otherwise the type not usable at all). And for the metadata of the storage item, we use `StorageHasher::METADATA`

Reply:
I like best what you just wrote: "The generic `Hasher` must implement the `StorageHasher` trait (or the type not usable at all). We use `StorageHasher::METADATA` for the metadata of the storage item."

Random observation: after Rust v1.48, writing `[`StorageHasher::METADATA`]` should be enough to get proper rust doc links to items that are in scope in the current file.

Reply:
yes done in https://github.com/paritytech/substrate/pull/7922/commits/11206a0f618efb51a6a8c63e42bdbe6d9168151b
I checked links are working